cpp/dolfinx/common/Scatterer.h

```diff
@@ -285,7 +285,9 @@ class Scatterer
                        MPI_Request& request)
   {
     assert(local_buffer.size() == _local_inds.size());
-    assert(_local_inds.size() <= local_data.size());
+    if (_local_inds.size() > 0)
```

Comment:
This isn't correct - the `assert` gets removed in non-debug mode, with the `if` then probably applied to the next statement.